lib/storage/src/content_manager/consensus_manager.rs

```diff
@@ -788,7 +788,8 @@ impl<C: CollectionContainer> ConsensusManager<C> {
             return Ok(false);
         };
 
-        if last_applied_index - first_entry.index < min_entries_to_compact {
+        debug_assert!(last_applied_index == 0 || last_applied_index >= first_entry.index - 1);
```

Comment:
Let's use a bit stricter condition:
```suggestion
        debug_assert!(last_applied_index >= first_entry.index || last_applied_index == 0);
```

Reply:
With #5900, WAL should *always* contain last applied entry, so `first_entry.index` should always be *same* as or *older/less* than `last_applied_index`.

These are valid WALs:
```
last_applied = 0, first_entry = 1
| 1 (unapplied) | 2 | 3 | ... |

last_applied = 43, first_entry = 43
| ... | 42 (applied, truncated) || 43 (applied) | 44 | ... |
```

But these are not:
```
last_applied = 42, first_entry = 43
| ... | 42 (applied, truncated) || 43 (unapplied) | 44 | ... |

last_applied = 41, first_entry = 43
| ... | 41 (applied, truncated) | 42 (unapplied, truncated) || 43 (unapplied) | 44 | ... |
```